File: src/main/java/com/ciazhar/authserver/repository/UserRepository.java
package com.ciazhar.authserver.repository;

import com.ciazhar.authserver.model.User;
import org.springframework.data.repository.PagingAndSortingRepository;

/**
 * Created by ciazhar on 6/14/17.
 */

/**
 * Repository untuk user
 */
public interface UserRepository extends PagingAndSortingRepository<User, String>
{
    User findByEmail(String email);
}

File: src/main/kotlin/com/ciazhar/authserver/repository/UserRepository.kt
package com.ciazhar.authserver.repository

import com.ciazhar.authserver.model.jpa.User
import org.springframework.data.repository.PagingAndSortingRepository

/**
 * Created by ciazhar on 6/14/17.
 */

/**
 * Repository untuk user
 */
interface UserRepository : PagingAndSortingRepository<User, String> {
    fun findByEmail(email: String): User
}
